source/stylesheets/components/_footer.scss

```diff
@@ -0,0 +1,116 @@
+//
+// Footer Component
+//
+// Based on the existing GOV.UK footer in GOV.UK Template, built with flexbox
+// and with tweaked behaviour for the OGL logo for the smaller viewport (when
+// allowing for the table of contents)
+
+@include exports("footer") {
+
+  .app-c-footer {
+    display: block;
+    margin-top: $govuk-spacing-scale-1;
+    padding: $govuk-gutter $govuk-gutter-half ($govuk-gutter * 2);
+    border-top: 1px solid $govuk-footer-border-top;
+    color: $govuk-footer-text;
+    background: $govuk-footer-background;
+    @include mq($from: 768px) {
+      padding: $govuk-gutter;
+    }
+
+    a:link,
+    a:visited {
+      color: $govuk-footer-link;
+    }
+
+    a:hover,
+    a:active {
+      color: $govuk-footer-link-hover;
+    }
+  }
+
+  .app-c-footer__heading {
+    margin-bottom: 0;
+    color: $govuk-black;
+  }
+
+  .app-c-footer__heading:nth-of-type(1) {
+    margin-top: 0;
+  }
+
+  .app-c-footer__list {
+    margin: 0;
+
+    li {
+      margin-bottom: 0;
+    }
+
+    a {
+      display: block;
+      padding: $govuk-spacing-scale-2 0 0;
+      font-size: 16px;
+    }
+  }
+
+  .app-c-footer__licence {
+    line-height: 1;
+    @include mq ($from: desktop) {
+      display: inline-block;
+      width: 75%;
+      vertical-align: bottom;
+    }
+  }
+
+  .app-c-footer__licence-logo {
+    display: block;
+    width: 41px;
+    height: 17px;
+    margin-top: -2px; // Align baseline with paragraph
+    overflow: hidden;
+    background-image: url("/images/open-government-licence_2x.png");
+    background-repeat: no-repeat;
+    background-size: 41px 17px;
+    text-indent: -999em;
+    @include mq($until: desktop) {
+      margin-bottom: $govuk-spacing-scale-3;
+    }
+    @include mq($from: desktop) {
+      display: inline-block;
+      margin-right: 12px;
+      vertical-align: top;
+    }
+  }
+
+  .app-c-footer__licence-description {
+    display: inline-block;
+    @include govuk-core-16;
+    margin: 0;
+  }
+
+  .app-c-footer__copyright {
+    @include govuk-core-16;
+    margin-top: $govuk-spacing-scale-5;
+    @include mq ($from: desktop) {
+      display: inline-block;
+      width: 25%;
+      margin-top: 0;
+    }
+  }
+
+  .app-c-footer__copyright-logo {
+    display: block;
+    min-width: 125px;
+    padding: 115px 0 0 0;
```

Comment:
Is there any way to avoid any of these magic numbers? Or to calculate them explicitly so we can understand where they came from? I assume this is to align the copyright logo at the bottom of the crest?

Reply:
I documented their provenance and calculate them when possible.